RHEnVision/provisioning-backend
# feat(kafka): setup, configuration, availability check

This adds sources availability checking via kafka backend worker.

To test this:

* Use scripts/kafka.setup.sh and kafka.start.sh to download, unpack and start up a small Kafka cluster (single process mode). You will need Java 8+ installed.
* Enable and configure Kafka. For dev setup all you need to do is `KAFKA_ENABLED=true`, when running kafka remotely then also `KAFKA_BROKERS=nuc.home.lan:9092`.
* Build and start `pbstatuser` process, that is "statusER". I could not figure out a better name, we already have a "worker" which is a generic job worker.
* Hit the `/availability_status` endpoint one or multiple times, there is a `.http` file for that.
* The API backend will enqueue and send to Kafka after few seconds. The statuser will pick them up (it can be delayed up to 10 seconds for performance reasons).
* It will currently only print a "trace" log, it does nothing. The actual implementation is outside of scope for this PR. @adiabramovitch was interested in helping with this, so here is where you can pick this up.

What was NOT tested:

* TLS
* SASL

I only tested with unencrypted plain connections (that is the default behavior). I will take a look into this in a separate PR where I will enable Kafka from the cluster. But "it should work".

More info on the design:

https://docs.google.com/document/d/1i18N1EwU3ccRWBAkqZFaJcGWTlY1rw6W-0gRt05f_Jc/edit
